SignalServiceKit/src/Account/TSAccountManager.m

```diff
@@ -93,7 +93,9 @@ - (void)resetForRegistration
             [transaction removeAllObjectsInCollection:TSAccountManager_UserAccountCollection];
         }];
     }
-    [[TSStorageManager sharedManager] resetSessionStore];
+    dispatch_async([OWSDispatch sessionStoreQueue], ^{
+        [[TSStorageManager sharedManager] resetSessionStore];
```

Comment:
PTAL @michaelkirk 

A small but important change.

Reply:
Actually, maybe this should dispatch sync.

Reply:
Good catch! Can you add an AssertIsOnSessionStoreQueue() to the underlying method?

I only use dispatch sync if it's difficult to add a completion handler, but it seems like adding a completion handler should be straight forward here.

Reply:
> Good catch! Can you add an AssertIsOnSessionStoreQueue() to the underlying method?

Done.

> Actually, maybe this should dispatch sync.

I've changed my mind - I don't think there's any point in blocking on this.